frontend/src/components/shared/StreamlitMarkdown/StreamlitMarkdown.tsx

```diff
@@ -303,9 +303,32 @@ export function LinkWithTargetBlank(props: LinkProps): ReactElement {
   )
 }
 
+function fixBodyForHeader(tag: string, body: string): string {
+  switch (tag) {
+    case "h1": {
+      body = `# ${body}`
+      break
+    }
+    case "h2": {
+      body = `## ${body}`
+      break
+    }
+    case "h3": {
+      body = `### ${body}`
+      break
+    }
+    default: {
+      throw new Error(`Unrecognized tag for header: ${tag}`)
+    }
+  }
+  return body
+}
+
 export function Heading(props: HeadingProtoProps): ReactElement {
   const { width } = props
-  const { tag, anchor, body } = props.element
+  let { body } = props.element
+  const { tag, anchor } = props.element
+  body = fixBodyForHeader(tag, body)
   const isSidebar = React.useContext(IsSidebarContext)
```

Comment:
All the above is fine, but it seems like this could be tighter. The thing that bothers me the most is the reassignment of `body`. It's not bad but less preferred in my eyes.

How about this?

Leave this as ` const { tag, anchor, body } = props.element` and....see below comment.

And the `HeadingWithAnchor`-> `RenderedMarkdown` is the following


```
<RenderedMarkdown
          source={fixBodyForHeader(tag, heading)} // Note I'm doing this on body but heading.
```

Based on this, I would just modify the name `fixBodyForHeader` to something meaningful like `makeMarkdownHeading(tag, headingText)`

(Please double check if this works as expected).